### ISSUE TITLE
Move, Reverse, etc. on LinkedList

### ISSUE THREAD

**Issue by Toxantron** [dotnet/corefx#27723]
Similar to #14338 we need the ability to manipulate a `LinkedList<T>`, most notably moving nodes within the collection. Another use case is to `Merge` one `LinkedList` into another one at a certain position (before / after a node). Technically the architecture of a double linked list and its nodes makes those feature quite easy to implement while offering good performance independent from list size.  

Unfortunately the current implementation of [`LinkedList<T>`](https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/Generic/LinkedList.cs) and its [`LinkedListNode<T>`](https://github.com/dotnet/corefx/blob/master/src/System.Collections/src/System/Collections/Generic/LinkedList.cs) is locked down like Fort-Knox, making it impossible to add the features in a derived type, even if I wanted to.  So I need to decide whether to implement the entire list myself or to work around the features. Most of the workarounds (e.g. Remove + AddBefore instead of MoveBefore) require creating new objects instead of changing references because most operations and members are internal or perform node validation when trying to insert nodes from a different collection.

Usefull additions to the API of `LinkedList`:
- Swap two nodes ([SO Thread](https://stackoverflow.com/questions/3473787/swapping-nodes-on-a-single-linked-list))
- MoveBefore and MoveAfter to move nodes within
- ReverseEnumerator: Currently the list implements IEnumerable and all its extensions only in forward direction. A few methods like Find make use of LinkedLists ability of reverse enumeration. For everything else I need to call `System.Linq.Reverse()` which copies the entire list to an array for the lack of a proper LinkedList implementation

**Comment by danmosemsft** [comment]
Hi @Toxantron thanks for the suggestion. Can you please format your top post in the API review standard format, as described here? 
https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md

Then an owner of this area can review your proposal.

**Comment by Toxantron** [comment]
Hi @danmosemsft I was not aware of the process. Maybe it wasn't around the last time I opened an issue. Is it better now?

**Comment by danmosemsft** [comment]
@Toxantron yes this looks great. Next step is the area owner gives feedback. Note that we are in API freeze for our 2.1 release. So there is no particular rush unfortunately. If it seems like this got lost feel free to ping me on the issue.
